### PR TITLE
[Feature][Connector-V2] Add web3j source connector

### PR DESCRIPTION
**Pull request by ic4y**
### Purpose of this pull request

Source connector for web3j. It is used to read data from the blockchain, such as block information, transactions, smart contract events, etc.  Currently, it supports reading block height data.

### Check list

* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
please fix ci

File: seatunnel-connectors-v2/connector-web3j/src/main/java/org/apache/seatunnel/connectors/seatunnel/source/Web3jSource.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                : Boundedness.UNBOUNDED;
+    }
+
+    @Override
+    public String getPluginName() {
+        return "Web3j";
+    }
+
+    @Override
+    public void prepare(Config pluginConfig) throws PrepareFailException {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-web3j/src/main/java/org/apache/seatunnel/connectors/seatunnel/source/Web3jSource.java]: remove this method

use factore create source

File: seatunnel-connectors-v2/connector-web3j/src/main/java/org/apache/seatunnel/connectors/seatunnel/source/Web3jSourceReader.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    @Override
+    public void close() throws IOException {
+        if (web3 != null) {
+            web3.shutdown();
+        }
+    }
+
+    @Override
+    public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-web3j/src/main/java/org/apache/seatunnel/connectors/seatunnel/source/Web3jSourceReader.java]: move to class field

**Comment by Hisoka-X** [comment]
Could you retrigger failed CI?

**Review by hailin0 — APPROVED** [review]
lgtm

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**